Some changes for Android and a new weapon overheating system.

madawe opened the pull request:
I did all my programming work in DemoControl.cs, HealthFlash.js and AutoFire.js

madawe2 committed:
Adjusted for Android and New Weapon Overheat Mechanic

In this commit I:
+ Configured the project to run on Android
+ Adjusted
the pause menu to properly fit on a mobile screen
+ Implemented
overheating mechanic so that if you fire continuously for too long the
weapon will jam until it cools down enough
+ Added a UI layer to add a
health bar and a heat bar using Unity sliders.
The .apk contains the
build after these changes.

madawe2 committed:
Adjusted for Android and New Weapon Overheat Mechanic

In this commit I:
+ Configured the project to run on Android
+ Adjusted
the pause menu to properly fit on a mobile screen
+ Implemented
overheating mechanic so that if you fire continuously for too long the
weapon will jam until it cools down enough
+ Added a UI layer to add a
health bar and a heat bar using Unity sliders.
The .apk contains the
build after these changes.

stelcheck commented:
This project is tab-indented; please make sure to follow the indentation of this project in your code submissions.

stelcheck commented:
Additionally, make sure to follow the same code standard consistently: for instance, when writing conditional statements:

```csharp
# Good
if (true) {
  // code
} 

# NG

if (true)
  // code

# NG

if(true){
} 
```

madawe commented:
I see you what you mean. I will make those changes.

madawe2 committed:
Adjusted formatting to the documents standards.

It is now tab-indented and uses consistent formatting with conditional
statements.

madawe2 committed:
Minor formatting change

madawe commented:
I updated AutoFire.js to have consistent formatting.
